### PR TITLE
Fix scaffolded files and add sanity checks to github workflow

### PR DESCRIPTION
**Pull request by mollux**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR addresses steps in the QA process that are easily forgotten:
- the `phpstan-baseline.neon` should only shrink, not altered (see [the commits after Nov 9, 2021](https://github.com/mautic/mautic/commits/4.x/phpstan-baseline.neon) for recent changes)
- the files used for scaffolding should be in sync with the original ones (e.g. the [original `.gitignore`](https://github.com/mautic/mautic/commits/4.x/.gitignore) has commits not present in the  [scaffolded `.gitignore`](https://github.com/mautic/mautic/commits/4.x/app/assets/scaffold/files/example.gitignore))

#### Steps to test this PR:

The tests in this PR fail intentionally to show the effect:
see https://github.com/mautic/mautic/pull/10857/checks
* the [`scaffolded files mismatch`](https://github.com/mautic/mautic/runs/5172853079?check_suite_focus=true#logs) fails when the scaffolded files that are not in sync with the original ones
* the [`PHPStan baseline changes`](https://github.com/mautic/mautic/runs/5172853097?check_suite_focus=true#logs) fails when there are unwanted changes to the `phpstan-baseline.neon` file

Once this makes sense to the reviewers of this PR, following steps are needed **before merging**:
* fix the scaffolded files
* revert the commit(s) that alter the `phpstan-baseline.neon` files

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10857"></a>

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -152,6 +152,27 @@ jobs:
...
             bin/php-cs-fixer fix --config=.php-cs-fixer.php -v --dry-run --using-cache=no --show-progress=dots --diff $cs_fix_files
           fi
+        elif [[ "${{ matrix.commands }}" == "scaffolded files mismatch" ]]; then
+          wget -q -O /tmp/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && chmod 755 /tmp/jq
+          /tmp/jq -r '.extra["mautic-scaffold"]["file-mapping"] | to_entries[] | "diff -q \(.key | sub("\\[(project|web)-root\\]";".")) app/\(.value)"' app/composer.json > diff_commands.sh
+          bash diff_commands.sh | tee /tmp/diff_command_output.txt
+          rm diff_commands.sh
+          if [[ $(wc -l </tmp/diff_command_output.txt) -ge 1 ]]; then
+            echo "some scaffolded files were not updated"
+            echo "Please apply the same changes in the files mentioned above"
```
Comment (escopecz) [inline .github/workflows/tests.yml:162]: Is there some command the dev can run to apply the changes? Or tell them somehow int he error message how to fix the problem?

Comment (mollux) [reply .github/workflows/tests.yml:162]: the files are mentioned in the output
(example of the current situation)

```
Files ./package-lock.json and app/assets/scaffold/files/package-lock.json differ
Files ./.gitignore and app/assets/scaffold/files/example.gitignore differ
Files ./upgrade.php and app/assets/scaffold/files/upgrade.php differ
```

I could add some line like 
```
please ensure the changes you made in the file(s) mentioned above are also present in their linked file in the `app/assets/scaffold/files` folder
```
I could parse that output and make example `cp` commands, but that's a bit too far IMO.
As some people only use an IDE and not CLI

Comment (escopecz) [reply .github/workflows/tests.yml:162]: Let's step back. I missed the reason behind the app/assets/scaffold folder, so I'm the great example of a person who has no idea what's going on. I don't have to play dumb but just myself :)

Why do we have 3 files that must be the same in 2 different locations? Can it be done with symlinks?

Comment (mollux) [reply .github/workflows/tests.yml:162]: Those files are needed for the composer based flow for Mautic, as they live outside of `mautic/core-lib`  
I assume the decision for copies of the files was discussed and made when setting up the composer flow.  

e.g. Drupal has the same issue: both tarball based installs and composer based installs are supported.  
But the composer based install can't rely on the files around the application (e.g. .htaccess, index.php)

so those files are changes twice, in their normal location, and in a location that can be used for scaffolding.
See e.g. [this commit in Drupal core](https://git.drupalcode.org/project/drupal/-/commit/d0be0af780017fa7af8471a5c93c7c1aa4a114a7): 
it changes the .htaccess file in 2 locations.

there is an [ongoing issue about this in Drupal core](https://www.drupal.org/project/drupal/issues/3075954) to remove the duplicate ones, and only rely on the scaffolded folder

Comment (escopecz) [reply .github/workflows/tests.yml:162]: I understand now. Thanks! The scaffolding check makes sense 👍 

I'll approve this PR once the CI is green.

Comment (mollux) [reply .github/workflows/tests.yml:162]: all tests green :)

**Review by escopecz — DISMISSED** [review]
Thanks! This is very helpful 👍

**Comment by kuzmany** [comment]
This also stop upgrade by `upgrade.php`? I do not see any mention in description , but see changelog

**Comment by mollux** [comment]
@kuzmany the description only mentions 1 example, but there are currently 3 files not in sync that are fixed in this PR.  
See [a previous build](https://github.com/mautic/mautic/runs/5172853079?check_suite_focus=true) for the 3 files that failed.

The test obviously passes now, so it's not easy to see indeed :)

**Comment by escopecz** [comment]
@kuzmany the changes here are just to sync the same files in the root dir with the same files in the `app/assets/scaffold/files` dir. So the changes in those file were already done in previous PRs but weren't copied to the scaffold.

**Review by kuzmany — APPROVED** [review]

**Review by kuzmany — DISMISSED** [review]

**Comment by kuzmany** [comment]
@mollux this is something in CI status what we you need check 

image

**Comment by mollux** [comment]
@kuzmany aha, I renamed 2 tests, but that's a bad idea, as those names are in the repo configuration

I reverted that in a54f3ca6c41c162152dc329654c6a6270a320d44

**Review by kuzmany — APPROVED** [review]

**Pull request merged**